File: app/src/main/java/com/rifafauzi/moviecatalogue/adapter/MovieAdapter.java
package com.rifafauzi.moviecatalogue.adapter;

import android.app.Activity;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.request.RequestOptions;
import com.rifafauzi.moviecatalogue.R;
import com.rifafauzi.moviecatalogue.model.Movies;
import com.rifafauzi.moviecatalogue.ui.DetailMovieActivity;

import java.util.ArrayList;
import java.util.List;

public class MovieAdapter extends RecyclerView.Adapter<MovieAdapter.MovieViewHolder> {

    private final Activity activity;
    private List<Movies> movies = new ArrayList<>();

    public MovieAdapter(Activity activity) {
        this.activity = activity;
    }

    public void setListMovies(List<Movies> listMovies) {
        if (listMovies == null) return;
        this.movies.clear();
        this.movies.addAll(listMovies);
    }

    @NonNull
    @Override
    public MovieAdapter.MovieViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_item, parent, false);
        return new MovieViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MovieAdapter.MovieViewHolder holder, int position) {
        Movies data = movies.get(position);
        holder.textViewTitle.setText(data.getTitle());
        holder.textViewDesc.setText(data.getOverview());
        holder.textViewDate.setText(data.getReleaseDate());
        Glide.with(holder.itemView.getContext())
                .load(Contract.LINK_IMAGE + data.getPosterPath())
                .apply(RequestOptions.placeholderOf(R.drawable.img_default_bg)
                        .error(R.drawable.ic_error))
                .into(holder.imageViewPoster);
        holder.itemView.setOnClickListener(v -> {
            Intent intent = new Intent(activity, DetailMovieActivity.class);
            intent.putExtra(DetailMovieActivity.EXTRA_MOVIE, ""+data.getId());
            activity.startActivity(intent);
        });
    }

    @Override
    public int getItemCount() {
        return movies.size();
    }

    class MovieViewHolder extends RecyclerView.ViewHolder {

        final TextView textViewTitle;
        final TextView textViewDesc;
        final TextView textViewDate;
        final ImageView imageViewPoster;

        MovieViewHolder(@NonNull View itemView) {
            super(itemView);
            textViewTitle = itemView.findViewById(R.id.movie_name);
            textViewDesc = itemView.findViewById(R.id.movie_desc);
            textViewDate = itemView.findViewById(R.id.movie_date);
            imageViewPoster = itemView.findViewById(R.id.movie_poster);
        }
    }
}

File: app/src/main/java/com/rifafauzi/moviecatalogue/adapter/MovieAdapter.kt
package com.rifafauzi.moviecatalogue.adapter

import android.app.Activity
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions
import com.rifafauzi.moviecatalogue.R
import com.rifafauzi.moviecatalogue.helper.local.entity.MoviesEntity
import com.rifafauzi.moviecatalogue.model.Movies
import com.rifafauzi.moviecatalogue.ui.detail.DetailMovieActivity

import java.util.ArrayList

class MovieAdapter(private val activity: Activity) : RecyclerView.Adapter<MovieAdapter.MovieViewHolder>() {
    private val movies = ArrayList<MoviesEntity>()

    fun setListMovies(listMovies: List<MoviesEntity>?) {
        if (listMovies == null) return
        this.movies.clear()
        this.movies.addAll(listMovies)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MovieViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.list_item, parent, false)
        return MovieViewHolder(view)
    }

    override fun onBindViewHolder(holder: MovieViewHolder, position: Int) {
        val data = movies[position]
        holder.textViewTitle.text = data.title
        holder.textViewDesc.text = data.overview
        holder.textViewDate.text = data.releaseDate
        Glide.with(holder.itemView.context)
                .load(Contract.LINK_IMAGE + data.posterPath)
                .apply(RequestOptions.placeholderOf(R.drawable.img_default_bg)
                        .error(R.drawable.ic_error))
                .into(holder.imageViewPoster)
        holder.itemView.setOnClickListener {
            val intent = Intent(activity, DetailMovieActivity::class.java)
            intent.putExtra(DetailMovieActivity.EXTRA_MOVIE, data.id.toString())
            activity.startActivity(intent)
        }
    }

    override fun getItemCount(): Int {
        return movies.size
    }

    inner class MovieViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        val textViewTitle: TextView = itemView.findViewById(R.id.movie_name)
        val textViewDesc: TextView = itemView.findViewById(R.id.movie_desc)
        val textViewDate: TextView = itemView.findViewById(R.id.movie_date)
        val imageViewPoster: ImageView = itemView.findViewById(R.id.movie_poster)

    }
}
